src/app-layer-ssl.c

```diff
@@ -143,11 +143,93 @@ static int SSLv3ParseHandshakeType(SSLState *ssl_state, uint8_t *input,
     switch (ssl_state->curr_connp->handshake_type) {
         case SSLV3_HS_CLIENT_HELLO:
             ssl_state->flags |= SSL_AL_FLAG_STATE_CLIENT_HELLO;
+
+            /* skip version */
+            input += 2;
+
+            /* skip random */
+            input += 32;
+
+            /* skip session id */
+            uint8_t session_id_length = *(input++);
+            input += session_id_length;
+
+            if ((input - initial_input) > input_len)
+                goto end;
+
+            /* skip cipher suites */
+            uint16_t cipher_suites_length = ntohs(*(uint16_t *)input);
+            input += cipher_suites_length + 2;
+
```

Comment:
Similarly here, the bounds are checked at line 157 to make sure there is enough input for the data already processed.  But we should check to make sure there is enough input data left before dereferencing at 161.